File: C++/119. Pascal's Triangle II.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> res;
        if(rowIndex < 0) return res;

        res.assign(rowIndex + 1, 0);
        for(int i = 0; i <= rowIndex; ++i) {
            if(i == 0) {
                res[0] = 1;
                continue;
            }
            for(int j = rowIndex; j >= 1; --j) {
                res[j] = res[j] + res[j - 1];
            }
        }
        return res;
    }
};

File: Algorithms/Java/119. Pascal's Triangle II.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> result = new LinkedList<>();
        long num = 1;
        int tem = 1;
        for(int i = 0 ; i <= rowIndex; i++){
            result.add(tem);
            num = num * (rowIndex-i) / (i + 1);
            tem=(int)num;
        }
        
        return result;
    }
}

// wrong soluton, it will overstack
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> result = new LinkedList<>();
        int num = 1;
        for(int i = 0; i <= rowIndex; i++){
            result.add(num);
            long tmp = num * (rowIndex-i) / (i + 1);
            num = (int) tmp;
        }
        return result;
    }
}